core/service.go

```diff
@@ -72,11 +71,18 @@ func (s *Service) start(e chan error) {
 				log.Error("core service", "error", err)
 				e <- err
 			}
+			e <- nil
 		case p2p.BlockAnnounceMsgType:
 			// get extrinsics by sending BlockRequest message
 			// process block
 		case p2p.BlockResponseMsgType:
 			// process response
+			err := s.ProcessBlock(msg[1:])
+			if err != nil {
+				log.Error("core service", "error", err)
+				e <- err
+			}
+			e <- nil
```

Comment:
I don't think we should send nil if there is an error

Reply:
the reason it sends nil is otherwise the other processes that have `e` will block on it forever if there's no error. if there's an error, it'll still get sent into `e` first

Reply:
line 206 in the test will block forever if we dont send nil

Reply:
I suppose [this pr](https://github.com/ChainSafe/gossamer/pull/334) is relevant here. It seems that if we don't really have a need to propagate errors ouf of the services to `Dot`, instead we can log it at the service level and possibly exit if we choose. That PR is will replace the channel with just an error, which should be one the occurs during the `Start` function not during the lifespan of the service. 

So I think the foreseeable issue here is whether we should exit if an error occurs here. Lmk what you think.

Feel free to leave this unless #334 gets merged first.

Reply:
I don't think we should exit, since stopping the message handler in the core service would effectively shut down any sort of block/tx processing. we could just log it for now